add items get api

festinalente91 opened the pull request:
Add Items get api

@fguy  @shberry16 @ovekyc @elegant651 @HewonJeong

festinalente91 committed:
add items get api

ovekyc commented on package.json:
```diff
@@ -76,7 +76,6 @@
     "browserify": "^13.1.0",
     "enzyme": "^2.4.1",
     "eslint-plugin-react": "^6.1.2",
-    "gulp": "^3.9.1",
```
bcz of duplicated both save and dev dependency

festinalente91 replied:
right :)

ovekyc commented on src/server/items/controllers.js:
```diff
@@ -2,18 +2,56 @@ import db from '../database';
 
 export default {
   getAll: (req, res, cb) => {
-    const result = {};
```
How about declare just const item = []
and req.send({ items: items });

festinalente91 replied:
that would be nice

ovekyc commented on src/server/items/controllers.js:
```diff
@@ -2,18 +2,56 @@ import db from '../database';
+      return cb();
     });
   },
-
```
please new line

festinalente91 replied:
oh, Trailing spaces not allowed by lint

fguy commented on src/server/items/controllers.js:
```diff
@@ -2,18 +2,56 @@ import db from '../database';
+    });
+  },
+  getById: (req, res, cb) => {
+    const key = 'item' + req.params.id;
```
Use template here:
const {id} = req.params;
const key = `item${id}`

ovekyc replied:
baeutiful

fguy commented on src/server/items/controllers.js:
```diff
@@ -2,18 +2,56 @@ import db from '../database';
+        if(err.notFound) {
+          res.status(400);
+          res.send({
+            error: "Item was not found."
```
Use single quotes for string literal. This must be caught by lint.

festinalente91 committed:
update lint

festinalente91 committed:
update lint

festinalente91 committed:
update lint

festinalente91 commented on package.json:
```diff
@@ -76,7 +76,6 @@
     "browserify": "^13.1.0",
     "enzyme": "^2.4.1",
     "eslint-plugin-react": "^6.1.2",
-    "gulp": "^3.9.1",
```
oh, Trailing spaces not allowed by lint

festinalente91 committed:
Remove parameter in cb()

Seems like no need to do something )

festinalente91 committed:
Merge branch 'master' into api/addItemGet

ovekyc commented:
LGTM

fguy commented on src/server/items/controllers.js:
```diff
@@ -2,18 +2,54 @@ import db from '../database';
+    })
+    .on('close', () => {
+      if (items.length !== 0) {
+        res.send({items: items});
```
nit; short hand of {items: items} is {items}. This must be caught by lint.

fguy commented:
LGTM besides of a nit comment. Please make sure your code doesn't occur lint warnings.

festinalente91 committed:
Update lint

{items:items} to {items}
https://github.com/airbnb/javascript#es6-object-concise

festinalente91 committed:
Merge branch 'master' into api/addItemGet

fguy commented:
LGTM
